url7895f5cfa7_edm92_defaultlogic_tgz-pJ8-be_fnord_DefaultLogic_AppJ8/src/be/fnord/util/logic/defaultLogic/RuleSet.java

```diff
@@ -0,0 +1,348 @@
+package be.fnord.util.logic.defaultLogic;
+
+import java.util.LinkedList;
+
+import be.fnord.util.logic.WFF;
+
+import com.merriampark.Gilleland.CombinationGenerator;
+
+// Todo add in extra functions for add remove rules
+
+public class RuleSet {
+
+	//
+	public LinkedList<String> addedConsequence = new LinkedList<String>();
+
+	private LinkedList<DefaultRule> rules = new LinkedList<DefaultRule>();
+
+	public void addRule(DefaultRule _rule) {
+		rules.add(_rule);
+	}
+
+	public LinkedList<String> applyRules(LinkedList<String> possibleExtensions,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.